onnx/defs/tensor/defs.cc

```diff
@@ -1191,6 +1191,12 @@ ONNX_OPERATOR_SET_SCHEMA(
             }
           }
 
+          // If transposing a scalar, we still want to infer a 0 dimension output shape
+          if (!shape.dim_size()) {
```

Comment:
I was wondering why the existing code wouldn't do the job already (with the zero-dimensional case being an instance of the more general n-dimensional case). I think what is missing is the initialization of the output-shape. I think the line
```cpp
getOutputShape(ctx, 0);
```
should be sufficient to do this initialization. May be worth trying.

Reply:
I agree it is the initialisation of the output shape - I wasn't quite sure how to initialise this so thanks